### PR TITLE
[V2] Code Conformance Cleanup (BREAKING CHANGES)

### PR DESCRIPTION
**Pull request by shbatm**
Clean-up code for conformance, reduce PyLint Errors, rename functions to snake_case from camelCase. Move folders to lowercase and update module to `pyisy` from `PyISY`.

Renames the various id variables `nid`, `pid` to a more consistent `address` property, which will be the same when accessing a Node, Group, Variable, Program, or Folder class.

Add specific exception types to error handlers.

### CODE REVIEW

File: pyisy/isy.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ class ISY:
     :ivar connected: Read only boolean value indicating if the class is
                      connected to the controller.
     :ivar log: Logger used by the class and its children.
-    :ivar nodes: :class:`~PyISY.Nodes.Nodes` manager that interacts with
+    :ivar nodes: :class:`~PyISY.nodes.Nodes` manager that interacts with
```
Comment (OverloadUT) [inline pyisy/isy.py]: Does PyISY and Nodes here need to change to pyisy and nodes?

File: pyisy/nodes/__init__.py [hunk 2]
```diff
@@ -55,12 +55,12 @@ class Nodes:
 
     This class can be used as a dictionary to
     navigate through the controller's structure to objects of type
-    :class:`~PyISY.Nodes.Node` and :class:`~PyISY.Nodes.Group` that represent
+    :class:`~PyISY.nodes.Node` and :class:`~PyISY.Nodes.Group` that represent
```
Comment (OverloadUT) [inline pyisy/nodes/__init__.py]: Same as previous comment

File: pyisy/programs/__init__.py [hunk 3]
```diff
@@ -1,407 +1,407 @@
...
+from .program import Program
+
+
+class Programs:
+    """
+    This class handles the ISY programs.
+
+    This class can be used as a dictionary
+    to navigate through the controller's structure to objects of type
+    :class:`~PyISY.programs.Program` and :class:`~PyISY.Programs.Folder`
```
Comment (OverloadUT) [inline pyisy/programs/__init__.py]: The second "Programs" in this one should be "programs" pretty sure

File: pyisy/variables/__init__.py [hunk 4]
```diff
@@ -1,207 +1,207 @@
...
+from .variable import Variable
+
+
+class Variables:
+    """
+    This class handles the ISY variables.
+
+    This class can be used as a     dictionary to navigate through the
+    controller's structure to objects of type
+    :class:`~PyISY.variables.Variable` that represent objects on the
```
Comment (OverloadUT) [inline pyisy/variables/__init__.py]: Same again

**Comment by shbatm** [comment]
I'll get them updated. Everything should be lowercase now except for the class names, [double-checked here](https://www.python.org/dev/peps/pep-0008/#package-and-module-names).

**Comment by shbatm** [comment]
Changes made. Also added a copy of the same `pylintrc` file from HA, not because I like it, but because I don't have anything better 😜 .

Currently outstanding:

```
************* Module pyisy.climate
climate.py:246:8: W0702: No exception type(s) specified (bare-except)
climate.py:259:16: W0702: No exception type(s) specified (bare-except)
climate.py:299:4: C0116: Missing function or method docstring (missing-function-docstring)
```

- Ignoring since climate will be retired soon.

```
************* Module pyisy.connection
connection.py:17:0: E0401: Unable to import 'requests.packages.urllib3.poolmanager' (import-error)
connection.py:61:16: E1101: Module 'requests.packages' has no 'urllib3' member (no-member)
connection.py:91:38: W1505: Using deprecated method encodestring() (deprecated-method)
```

- Have a staged request for dropping Python 2 support. This will be resolved there.

```
************* Module pyisy.nodes.node
nodes/node.py:223:-1: W0105: String statement has no effect (pointless-string-statement)
```

- Have a staged pull request which will drop this comment.

```
************* Module pyisy.programs.__init__
programs/__init__.py:244:2: W0511: TODO: Unpack into proper properties. (fixme)
```

- Yep. On the list.

File: pyisy/nodes/nodebase.py [hunk 5]
```diff
@@ -135,6 +135,7 @@ def send_cmd(self, cmd, val=None, uom=None, query=None):
         )
 
         # Calculate hint to use if status is updated
+        # pylint: disable=protected-access
```
Comment (shbatm) [inline pyisy/nodes/nodebase.py]: Adding this pylint override because Typing issues with VarEvents (see #60).

**Comment by shbatm** [comment]
Update 3/10: Testing on beta channel with these changes seems OK. Let me know if you see anything else you want to include in this PR. Then I think I have 2 more small ones to be caught up with the Beta channel.

**Comment by shbatm** [comment]
> Then I think I have 2 more small ones to be caught up with the Beta channel.

OK. I lied. You're going to have about 5 more after the current set.

**Pull request merged**